src/Endpoints/class-base-api-proxy.php

```diff
@@ -172,4 +174,89 @@ protected function get_data( WP_REST_Request $request, bool $require_api_secret
 			'data' => $this->generate_data( $response ), // @phpstan-ignore-line.
 		);
 	}
+
+	/**
+	 * Extracts the post data from the passed object.
+	 *
+	 * Should only be used with endpoints that return post data.
+	 *
+	 * @since 3.10.0
+	 *
+	 * @param stdClass $item The object to extract the data from.
+	 *
+	 * @return array<string, mixed> The extracted data.
```

Comment:
Adding proper typing here will be helpful

Reply:
You mean something like `@return array<string, string|bool|int|null>`, or you mean using PHPStan types? The reasons I don't use PHPStan types have been discussed in another PR. We use `@return array<string, mixed>` regularly in our codebase, but we can see if it would be good to change that with more specific types.

Reply:
I mean define whole structure of the return object [Parsely_Options](https://github.com/Parsely/wp-parsely/blob/c3fda7876e2f43a382cdd230f6115cd2b32e151e/src/class-parsely.php#L23)

Reply:
This would need a new PHPStan type.

I've expressed why I consider them problematic [here](https://github.com/Parsely/wp-parsely/pull/1816#discussion_r1304094701). My opinion remains the same currently and I don't wish to invest further time in that, unless there's a suggestion that fixes the shortcomings mentioned in that thread.